stout/flags.cc

```diff
@@ -21,32 +21,86 @@ namespace stout::flags {
 void Parser::AddAllOrExit(google::protobuf::Message* message) {
   const auto* descriptor = message->GetDescriptor();
 
+  auto exit_on_missing_name =
+      [](const google::protobuf::FieldDescriptor* field) {
+        std::cerr
+            << "Missing at least one flag name in 'names' for field '"
+            << field->full_name() << "'"
+            << std::endl;
+        std::exit(1);
+      };
+
+  auto exit_on_missing_help =
+      [](const google::protobuf::FieldDescriptor* field) {
+        std::cerr
+            << "Missing flag 'help' for field '"
+            << field->full_name() << "'"
+            << std::endl;
+        std::exit(1);
+      };
+
   for (int i = 0; i < descriptor->field_count(); i++) {
     const auto* field = descriptor->field(i);
 
-    const auto& flag = field->options().GetExtension(stout::v1::flag);
+    // We need this descriptor for the subcommand's logic.
+    const google::protobuf::OneofDescriptor* real_oneof_field =
+        field->real_containing_oneof();
+
+    // Check if the current field of the message is in 'oneof'.
+    if (real_oneof_field != nullptr) {
+      // Check if the name of 'oneof' is 'subcommand'. If no -> exit with
+      // an error.
+      if (real_oneof_field->name() != "subcommand") {
+        std::cerr << "'oneof' field must have 'subcommand' name. "
+                  << "Other names are illegal"
+                  << std::endl;
+        exit(1);
+      } else {
+        // Subcommands must have stout.v1.subcommand extension.
+        if (!field->options().HasExtension(stout::v1::subcommand)) {
+          std::cerr << "Every field of the 'oneof subcommand' must "
+                       "have (stout.v1.subcommand) extension"
+                    << std::endl;
+          exit(1);
+        } else {
+          // Check for missing 'names' and 'help'
+          // in (stout.v1.subcommand) extension.
+          const auto& subcommand =
+              field->options().GetExtension(stout::v1::subcommand);
+
+          if (subcommand.names().empty()) {
+            exit_on_missing_name(field);
```

Comment:
Ah, I see why you want to reuse this but the error message now says _flag_ instead of _subcommand_!

Reply:
I've change this if statement in the [PR](https://github.com/3rdparty/stout-flags/pull/18) for parsing pos args.

Reply:
This is still a problem though because it prints out the word "flag" instead of "subcommand". You have three things, flags, subcommands, and positional arguments. By reusing this lambda helper the error message here is "Missing at least one flag name in 'names' for field ..." but what you want it to say is "Missing at least one subcommand name in 'names' for field ...". So just don't reuse the lambda helper but copy the `std::cerr` and `std::exit()` calls.